examples/convert_mri_and_trans.py

```diff
@@ -24,19 +24,145 @@
-#
-#
-#
+
+import os
+import os.path as op
+import shutil as sh
+
+import numpy as np
+import mne
+from mne.datasets import sample
+from mne.transforms import apply_trans
+import nibabel as nib
+import matplotlib.pyplot as plt
+from nilearn.plotting import plot_anat
+
+from mne_bids import (write_raw_bids, make_bids_basename, write_anat,
+                      get_head_mri_trans)
+from mne_bids.utils import print_dir_tree, _extract_landmarks
+
+###############################################################################
+# We will be using the `MNE sample data <mne_sample_data_>`_ and write a basic
+# BIDS dataset. For more information, you can checkout the respective
+# :ref:`example <ex-convert-mne-sample>`.
+
+data_path = sample.data_path()
+event_id = {'Auditory/Left': 1, 'Auditory/Right': 2, 'Visual/Left': 3,
+            'Visual/Right': 4, 'Smiley': 5, 'Button': 32}
+raw_fname = op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw.fif')
+events_data = op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw-eve.fif')
+output_path = op.abspath(op.join(data_path, '..', 'MNE-sample-data-bids'))
+if op.exists(output_path):
+    sh.rmtree(output_path)
+raw = mne.io.read_raw_fif(raw_fname)
+sub = '01'
+ses = '01'
+task = 'audiovisual'
+run = '01'
+bids_basename = make_bids_basename(subject=sub, session=ses, task=task,
+                                   run=run)
+write_raw_bids(raw, bids_basename, output_path, events_data=events_data,
+               event_id=event_id, overwrite=True)
+
+# Print the directory tree
+print_dir_tree(output_path)
+
+###############################################################################
+# Now let's assume that we have also collected some T1 weighted MRI data for
+# our subject. And furthermore, that we have already aligned our coordinate
+# frames (as described HERE) and obtained a transformation matrix
+# :code:`trans`.
+
+# Get the path to our MRI scan and covert it to NIfTI format
+# For that, we use nibabel, reading in the compressed mgh file, and saving it
+# as a compressed NIfTI file
+t1_mgh_fname = op.join(data_path, 'subjects', 'sample', 'mri', 'T1.mgz')
+t1_mgh = nib.load(t1_mgh_fname)
+t1_nii_fname = op.join(data_path, 'subjects', 'sample', 'mri', 'T1.nii.gz')
+nib.save(t1_mgh, t1_nii_fname)
+
+# Load the transformation matrix and show what it looks like
+trans_fname = op.join(data_path, 'MEG', 'sample',
+                      'sample_audvis_raw-trans.fif')
+trans = mne.read_trans(trans_fname)
+print(trans)
+
+###############################################################################
+# We can save the MRI to our existing BIDS directory and at the same time
+# create a JSON sidecar file that contains metadata, we will later use to
+# retrieve our transformation matrix :code:`trans`.
+
+# We use the write_anat function
+write_anat(bids_root=output_path,  # point to the BIDS dir we wrote earlier
+           subject=sub,
+           t1w=t1_nii_fname,  # path to the MRI scan in NIfTI format
```

Comment:
I feed that t1w should accept any format readable by nibabel and an instance of 3D niimg-like object (as interpreted by nilearn). It would simplify your example as you would not need to convert anything to use write_anat

Reply:
yes, it's much better that way - thanks! https://github.com/mne-tools/mne-bids/pull/211/commits/5233a6f8c648a88dc794de125cca0c4b029fdaff